doc/1/plugins/plugin-context/customize-errors/index.md

```diff
@@ -0,0 +1,61 @@
+---
+code: true
+type: page
+title: customize errors
+---
+
+# Customize errors
+
+When creating your own Kuzzle plugin, you can use the `errorsManager` to have customizable errors.
+
+In order to define it, you have to write them inside the [manifest.json](https://docs.kuzzle.io/core/1/plugins/guides/manual-setup/prerequisites/#manifest-json) in a `errors` field.
+
+Your manifest would be something like :
+```
+{
+    "name": "kuzzle-plugin-xxx",
+    "kuzzleVersion": ">=1.0.0 <2.0.0",
+    "privileged": false,
+    "errors": {
+        "some_error": {
+            "code": 1,
+            "message": "Some error occurred: %s",
+            "class": "BadRequestError"
+	},
+        "some_other_error": {
+            "code": 2,
+            "message": "Some other error occurred: %s",
+            "class": "ForbiddenError"
+	}
+    }
+}
+```
+
+It is exposed as `errorsManager` in the [PluginContext](https://docs.kuzzle.io/core/1/plugins/plugin-context/accessors/intro/).
+The `errorsManager` provides two functions:
+- To throw : `context.errorsManager.throw(error, placeholders);`.
+- To get the built error: `context.errorsManager.get(error, placeholders);`
+
+Also, you can give more precision by giving a [`subdomain`](https://docs.kuzzle.io/core/1/plugins/plugin-context/errors/kuzzleerror/) to your plugin in you [kuzzlerc file](https://docs.kuzzle.io/core/1/plugins/guides/manual-setup/config/). It would be like :
+
+```
+{
+  "plugins": {
+    "foobar-plugin": {
+      "option_1": "option_value",
+      "option_2": "option_value",
+      "codes": {
```

Comment:
I didn't pay attention in the implementation PR, but I find this very ugly and redundant.

... shouldn't this be `"subdomain": 42`?

Also, now that I see this documentation... there is something that I don't like: this configuration is normally interpreted by plugins themselves, but here we have introduced a reserved word for our own usage. We need to isolate it from the rest.

One possible way to do that would be, perhaps, to prefix it with an underscore, as we do with other particular properties in our API?

Reply:
I prefere prefixing it like: `_pluginCodes` or just `_codes`.
I find it more explicit than a `subdomain` property here . I think it's more clear and easy to understand for developers.

Reply:
Ok for an explicit name... How about `_pluginErrorCode`?